src/test/lib/storage/buffer/frame_test.cpp

```diff
@@ -0,0 +1,150 @@
+#include "base_test.hpp"
+#include "storage/buffer/frame.hpp"
+
+namespace hyrise {
+
+class FrameTest : public BaseTest {};
+
+TEST_F(FrameTest, TestStateTransitions) {
+  auto frame = Frame{};
+  EXPECT_EQ(Frame::state(frame.state_and_version()), Frame::EVICTED);
+  EXPECT_EQ(Frame::version(frame.state_and_version()), 0);
+
+  auto old_state_and_version = frame.state_and_version();
+  EXPECT_TRUE(frame.try_lock_exclusive(old_state_and_version));
+  EXPECT_EQ(Frame::version(frame.state_and_version()), 0);
+  EXPECT_EQ(Frame::state(frame.state_and_version()), Frame::LOCKED);
+  EXPECT_FALSE(frame.try_lock_exclusive(old_state_and_version));
+  EXPECT_FALSE(frame.try_lock_shared(old_state_and_version));
+
+  frame.unlock_exclusive();
+  EXPECT_EQ(Frame::state(frame.state_and_version()), Frame::UNLOCKED);
+  EXPECT_EQ(Frame::version(frame.state_and_version()), 1);
+  EXPECT_TRUE(frame.is_unlocked());
+
+  // Lock 3 times in shared mode
+  old_state_and_version = frame.state_and_version();
+  EXPECT_TRUE(frame.try_lock_shared(old_state_and_version));
+  EXPECT_EQ(Frame::version(frame.state_and_version()), 1);
+  EXPECT_EQ(Frame::state(frame.state_and_version()), 1);
```

Comment:
Replacing `1` with `SINGLE_LOCKED_SHARED` might demonstrate its usage and descriptiveness.